makoctl

```diff
@@ -130,9 +136,50 @@ case "$1" in
 "reload")
 	call Reload
 	;;
-"set-mode")
+"set-mode") # deprecated
 	call SetMode "s" "$2"
 	;;
+"mode")
+	shift 1
+	require_jq
+	modes="$(call ListModes | jq '.data[0]')"
+	add_remove_flag=0
+	set_flag=0
+	while getopts a:r:s name; do
+		case "$name" in
+		a)
+			add_remove_flag=1
+			modes="$(echo "$modes" | jq --arg mode "$OPTARG" '. += [$mode]')"
+			;;
+		r)
+			add_remove_flag=1
+			modes="$(echo "$modes" | jq --arg mode "$OPTARG" 'del(.[] | select(. == $mode))')"
+			;;
+		s)
+			set_flag=1
+			;;
+		?)
+			exit 1
+		esac
+	done
+	if [ "$add_remove_flag" = 1 ] && [ "$set_flag" = 1 ]; then
+		echo >&2 "makoctl: -a/-r and -s cannot be used together"
+		exit 1
+	fi
+	if [ "$set_flag" = 1 ]; then
+		shift $(($OPTIND - 1))
+		modes="$(jq -n '$ARGS.positional' --args "$@")"
+	fi
+	if [ "$add_remove_flag" = 1 ] || [ "$set_flag" = 1 ]; then
+		modes="$(echo "$modes" | jq '. | unique')"
+		modes_len="$(echo "$modes" | jq '. | length')"
+		modes_args=$(echo "$modes" | jq -r '@sh')
+		# Required to behave properly when mode names contain spaces
+		eval set -- $modes_args
+		call SetModes "as" "$modes_len" "$@"
```

Comment:
If the last mode is removed, should we automatically set the modes list to just `default`?

Reply:
The only downside of getting rid of it that I can think of is that there would be no way to write a config block that takes effect when there are zero modes active, since `mode=` would try to match a mode named empty string. I don't know if there is actually a usecase for this, but perhaps it would be more useful now that multiple modes are supported than it would have been before?

We could of course also special-case `mode=` to select for the empty list.

Reply:
I don't see a use-case for this TBH.

Reply:
Done. BTW, if someone comes up with a good use-case for a default mode, we could add something like `mako --initial-mode default` to set the initial list of modes.